node/silkworm/stagedsync/stage_hashstate.cpp

```diff
@@ -26,9 +26,7 @@
 namespace silkworm::stagedsync {
 
 StageResult HashState::forward(db::RWTxn& txn) {
-    if (is_stopping()) {
-        return StageResult::kAborted;
-    }
+    continue_or_throw();
```

Comment:
Hmm. I'm not sure about throwing instead of returning an error code. IMHO the previous version was better. It's not that I'm against exceptions, but we should be consistent. If we rely on exceptions, then `HashState::forward` should be `void`, not return `StageResult`.